### PR TITLE
Yazan habit creation

### PR DESCRIPTION
**Pull request by crankdank**
Added my habits activity, a fragment for creating habits, and only the ability to add in habits. Editing habits and removing habits remain to be implemented, as well as a habit validity check

### CODE REVIEW

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                    calendarView.setVisibility(View.VISIBLE);
+                else
+                    calendarView.setVisibility(View.GONE);
+                //calendarView.setVisibility(calendarView.getVisibility() == View.GONE ? View.VISIBLE : View.GONE);
+            }
+        });
+
+        Bundle args = getArguments();
+        if (args != null) {
+            Habit Retrieved_Habit = (Habit)args.getSerializable("habit");
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java:100]: I think we should keep our variable naming convention consistent, others use lower camel case with no underscores

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 32);
+                            checkBox = view.findViewById(R.id.sundayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 64);
+                            listener.onOkPressed(Retrieved_Habit, name,reason, date, weekdays);
+                        }
+                    }).create();
+        }
+        else {
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java:142]: This chunk of code under here seems identical to the code above starting from line 103. I don't think there's a reason to lock it into a conditional then. Perhaps just put what's different (lines 100-102) inside the conditional and only have one instance of this giant chunk of code.

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/MyHabits.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class MyHabits extends AppCompatActivity implements HabitFragments.OnFragmentInteractionListener{
+    private RecyclerView recyclerView;
+    List<Habit> modelList;
+    private Button removeButton;
+    private Button cancelButton;
+    private RecyclerViewAdapter adapter2;
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/MyHabits.java:19]: I can't seem to find another adapter item, so why is this the second? Perhaps a more descriptive name of what this adapter is.

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/MyHabits.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+         retrieved_habit.setStartDate(startDate);
+         retrieved_habit.setReason(reason);
+         retrieved_habit.setWeekdays(weekdays);
+         adapter2.notifyDataSetChanged();
+    }
+
+    @Override
+    public void onPointerCaptureChanged(boolean hasCapture) {
+
+    }
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/MyHabits.java:83]: Why is there an empty function here?

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/RecyclerViewAdapter.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    notifyItemChanged(getAdapterPosition());
+
+                }
+            });
+            editButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Habit habit = habitList.get(getAdapterPosition());
+                    HabitFragments Edit_frag = new HabitFragments().newInstance(habit);
+                    //Edit_frag.show(this,"EDIT_CITY");
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/RecyclerViewAdapter.java:73]: please cleanup all these commented lines of code

File: CMPUT_301_Project/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -2,4 +2,14 @@
     <string name="app_name">CMPUT_301_Project</string>
     <!-- TODO: Remove or change this placeholder text -->
     <string name="hello_blank_fragment">Hello blank fragment</string>
+    <string name="title_activity_temp">temp</string>
+    <!-- Strings used for fragments for navigation -->
+    <string name="first_fragment_label">First Fragment</string>
+    <string name="second_fragment_label">Second Fragment</string>
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/res/values/strings.xml:8]: what are these? more descriptive names if these are necessary?

File: CMPUT_301_Project/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -2,4 +2,14 @@
...
+    <string name="next">Next</string>
+    <string name="previous">Previous</string>
+
+    <string name="hello_first_fragment">Hello first fragment</string>
+    <string name="hello_second_fragment">Hello second fragment. Arg: %1$s</string>
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/res/values/strings.xml:13]: same question with these as above

File: ProjectPart2.txt [hunk 8]
```diff
@@ -0,0 +1,90 @@
+Product Backlog for habit app demand:
```
Comment (ccchoi2) [inline ProjectPart2.txt:1]: Why does this txt file exist? If it's a test file, please keep it local to your machine

**Review by ccchoi2 — CHANGES_REQUESTED** [review]

**Review by adiep2 — APPROVED** [review]
Looks good just need to address Calvin's comments.  Will approve since its just comment and variable name changes and also so we don't need to wait on it later

**Review by ccchoi2 — APPROVED** [review]

**Pull request merged**